src/main/java/org/peergos/protocol/dht/Kademlia.java

```diff
@@ -41,7 +41,14 @@ public void setAddressBook(AddressBook addrs) {
 
     public int bootstrapRoutingTable(Host host, List<MultiAddress> addrs, Predicate<String> filter) {
         List<String> resolved = addrs.stream()
-                .flatMap(a -> DnsAddr.resolve(a.toString()).stream())
+                .parallel()
+                .flatMap(a -> {
+                    List<String> record = new ArrayList<>();
```

Comment:
Can we just return directly in the two cases with.stream and Stream.empty() and avoid a mutable variable?